### DISCUSSION TITLE
Unable to attach or mount volumes on Kernel 6.2.0

### DISCUSSION THREAD

**Discussion by bfeher-kx** [rook/rook#13199] [Q&A]
We have lately received an Azure AKS node pool with Ubuntu 22.04 HWE kernel 6.2.0-1014-azure (usually meant for special hardware eg.: GPU enabled nodes) and rook-cephfs RWX volumes are no longer working.

The exact same configuration works as expected on nodes with Ubuntu 22.04 Generic kernel 5.15.0-1051-azure.

INFO:
Kubernetes version: 1.26.3
Rook version: 1.11.2
Ceph version: 17.2.5

LOGS:
`    health: HEALTH_WARN
            1 clients failing to respond to capability release
            1 MDSs report slow requests
`
Pod level:
`Warning  FailedMount  64s (x2488 over 3d21h)  kubelet  (combined from similar events): Unable to attach or mount volumes: unmounted volumes=[x], unattached volumes=[ x, y, z configmaps, secrets and RWX volumes]: timed out waiting for the condition`

MDS:
`log_channel(cluster) log [WRN] : 9 slow requests, 0 included below; oldest blocked for > 676551.015794 secs`

Snippet from ops_in_flight dump:
`"event": "failed to rdlock, waiting"
"event": "failed to xlock, waiting"
`
We are suspecting here mounting issues, Ceph backend seems to be fine. The deployment with this state is still running, It would be great to get know what is happening here and how we can react if similar scenario happens.

Any suggestions are warmly welcomed!

**Comment by microyahoo** [comment]
What's the full output of `ceph -s`? Have a look at `ceph daemon mds.<id> objecter_requests` while it is
stuck to see if there are OSD requests that are stuck from the MDS.

**Reply by bfeher-kx** [reply]
Thanks for the reply! Here are the outputs:

```
 cluster:
    id:     1f5b118a-6af4-4a30-a3da-8f1fbf1af12f
    health: HEALTH_WARN
            1 clients failing to respond to capability release
            1 MDSs report slow requests
            1 mgr modules have recently crashed (I have manually killed it)

  services:
    mon: 3 daemons, quorum a,b,c (age 22h)
    mgr: a(active, since 4d)
    mds: 1/1 daemons up, 1 hot standby
    osd: 10 osds: 10 up (since 11d), 10 in (since 11d)

  data:
    volumes: 1/1 healthy
    pools:   4 pools, 81 pgs
    objects: 397 objects, 24 MiB
    usage:   600 MiB used, 5.9 TiB / 5.9 TiB avail
    pgs:     81 active+clean

  io:
    client:   853 B/s rd, 1 op/s rd, 0 op/s wr
```

`ceph daemon mds.myfs-a objecter_requests
{
    "ops": [],
    "linger_ops": [],
    "pool_ops": [],
    "pool_stat_ops": [],
    "statfs_ops": [],
    "command_ops": []
}
`
It seems request are empty but the pods are still unable to start, any other ideas?

Thanks!

**Reply by microyahoo** [reply]
does umount then mount from the client clear the warning?  or try to release client session like [KB](https://www.suse.com/support/kb/doc/?id=000019628)

**Reply by bfeher-kx** [reply]
Thanks for your suggestion, the detailed status showed the affected node, I've restarted it and the pending PODs were able to start. 
I've had a feeling it wouldn't be a fix as we were able to reproduce this in 5 individual AKS deployments.
I've redeployed our deployments in this AKS and its stuck again, now 4 nodes are failing to respond to capability release.
At least this is a good indicator to point on the Cephfs client misbehaving on 6.2.0

**Reply by microyahoo** [reply]
Currently we have not met this problem on Centos 7.9 and 8.5, you can send an email to ceph mailing list or file a issue to ceph tracker.
